### ISSUE TITLE
required Single-Option type includes redundant empty string

### ISSUE THREAD

**Issue by blindfish3** [dohomi/storyblok-generate-ts#8]
In Storyblok I have a Single-Option field that is required and has a default value set.  It's for a form button so the options are `button` and `submit`.  When I run `storyblok-generate-ts` the definition for this field is as follows:

```ts
buttonType: "" | "button" | "submit";
```

In this case the empty string is not a valid option for this field and should not be included in the generated type.

**Comment by dohomi** [comment]
@blindfish3 this PR made the empty string possible and I haven't encountered any issue with that:
https://github.com/dohomi/storyblok-generate-ts/pull/5

**Comment by blindfish3** [comment]
OK - I now see how to exclude the empty string; but It doesn't really make a lot of sense to me.

So, to resolve this issue you can tick the "Hide empty option" checkbox in Storyblok and then the empty string is removed from the type.  In the interface that option hides any empty option that is added.

The problem I have with that solution is that  - in Storyblok - that checkbox only has an effect if an empty option is included.  In my case there is no empty option; so I don't need to tick the box and I don't expect to see one in the types:

![image](https://user-images.githubusercontent.com/11975948/177742158-81cd3501-0fbf-435a-a4d8-a4b5281da3fa.png)

Here I would expect `buttonType:  "button" | "submit";`

![image](https://user-images.githubusercontent.com/11975948/177742819-d10c35b4-426c-40a3-b144-76f45fd5f3d4.png)

Here the empty string would be expected: `buttonType: "" | "button" | "submit";`

![image](https://user-images.githubusercontent.com/11975948/177743310-c58d2e25-e121-46bb-bd47-c884573c7298.png)

Where there is an empty option but "Hide empty option" is ticked I would again expect `buttonType:  "button" | "submit";`

Does that make sense?

**Comment by blindfish3** [comment]
> I think your point of view just collides with the PR from other users who want to have an empty option string

I don't think so: I'm just saying that the empty string option should only find its way into the type when it is a possible input value; and that isn't the current behaviour.
Anyway for now I just have to remember to tick "Hide empty option".  I'll have to check if I can find some time to raise a PR.
